src/views/Products.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import ProductsSidebar from '../components/ProductsSidebar'
+import ProductsJson from '../mocks/en-us/products.json'
+import '../styles/Products.scss'
+import ProductItem from '../components/FeaturedProducts/ProductItem'
+import ProductsPagination from '../components/ProductsPagination'
+import Spinner from '../components/Spinner'
+import { Link, useNavigate, useSearchParams } from 'react-router-dom'
+import { useProducts } from '../utils/hooks/useProducts'
+import { useSelector } from 'react-redux'
+
+export default function Products () {
+  const navigate = useNavigate()
+  const [searchParams] = useSearchParams()
+  const [filters, setFilters] = useState<Array<string>>([])
+  const { products, getProducts } = useProducts()
+  const {
+    data: categories,
+    isLoading
+  } = useSelector((state: any) => state.categories)
+
+  const handleCheckInput = (id: string|null, value: boolean = true) => {
+    const categoryToBeHandled: any = document.getElementById(`category-${id}`)
```

Comment:
there a react why of doing this `document.getElementById('category-${id}')`, modifying the DOM directly is not a good idea, the component definition could live in a file and the part of the code that changes the component could be anywhere in the project